### PR TITLE
Remove some unnecessary <iostream> includes from headers

### PR DESCRIPTION
**Pull request by peterbell10**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #106915
* __->__ #106914

In almost all cases this is only included for writing the output formatter, which
only uses `std::ostream` so including `<ostream>` is sufficient.

The istream header is ~1000 lines so the difference is non-trivial.

cc @jgong5 @mingfeima @XiaobingSuper @sanchitintel @ashokei @jingxu10 @mcarilli @ptrblck @leslie-fang-intel @EikanWang @voznesenskym @penguinwu @anijain2305 @Guobing-Chen @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @ipiszy @chenyang78 @aakhundov

### CODE REVIEW

File: .github/ci_commit_pins/xla.txt [hunk 1]
```diff
@@ -1 +1 @@
-709c75a04d9b35d83a9509e1534a8aa2046b8912
+c3c16ccac41cb2db6ba88fb31342f4af62c7e15a
```
Comment (peterbell10) [inline .github/ci_commit_pins/xla.txt:1]: This is so we include pytorch/xla#5434

**Review by lezcano — APPROVED** [review]
Did you apply these by hand, or is it possible to automatise these?

**Comment by lezcano** [comment]
Talking from memory here, but I think that include-what-you-use does do the iosfwd checks.

**Comment by lezcano** [comment]
It seems that it suggests the right thing... sometimes https://github.com/include-what-you-use/include-what-you-use/issues/683

**Comment by kit1980** [comment]
@peterbell10 I'll help with relanding this.

**Comment by kit1980** [comment]
@kit1980 has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D48670403).

**Comment by kit1980** [comment]
@kit1980 has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D48670403).

**Comment by kit1980** [comment]
@kit1980 has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D48670403).

**Comment by kit1980** [comment]
@kit1980 has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D48670606).

**Comment by kit1980** [comment]
Looks like I missed some issues when re-landing first time...

**Comment by peterbell10** [comment]
My branches seem to have been deleted, so I've opened a new stack starting at #108149. Also switched the PR order so hopefully the pch update doesn't need to be reverted again.

**Pull request closed without merging**